tests/integration-tests/backup_policy.tftest.hcl

```diff
@@ -0,0 +1,87 @@
+mock_provider "azurerm" {
+  source = "./azurerm"
+}
+
+run "setup_tests" {
+  module {
+    source = "./setup"
+  }
+}
+
+run "create_blob_storage_policy" {
+  command = apply
+
+  module {
+    source = "../../infrastructure"
+  }
+
+  variables {
+    vault_name = run.setup_tests.vault_name
+  }
+
+  # Check that the id is as expected
```

Comment:
The tests are self-documenting, you could remove the comments wrapped around the tests without compromising clarity